storm-server/src/main/java/org/apache/storm/localizer/AsyncLocalizer.java

```diff
@@ -144,8 +148,19 @@ public Void call() throws Exception {
                 File tr = new File(tmproot);
                 try {
                     downloadBaseBlobs(tr);
+                    if (_assignment.is_set_total_node_shared()) {
+                        File sharedMemoryDirTmpLocation = new File(tr, "shared_by_topology");
+                        //We need to create a directory for shared memory to write to (we should not encourage this though)
+                        Path path = sharedMemoryDirTmpLocation.toPath();
+                        Files.createDirectories(path);
+                    }
                     _fsOps.moveDirectoryPreferAtomic(tr, _stormRoot);
-                    _fsOps.setupStormCodeDir(ConfigUtils.readSupervisorStormConf(_conf, _topologyId), _stormRoot);
+                    Map<String, Object> topoConf = ConfigUtils.readSupervisorStormConf(_conf, _topologyId);
+                    _fsOps.setupStormCodeDir(topoConf, _stormRoot);
+                    if (_assignment.is_has_node_shared_memory()) {
+                        File sharedMemoryDir = new File(_stormRoot, "shared_by_topology");
```

Comment:
Okay. I was just thinking it might be helpful if the topology could get the directory path (or _stormRoot), so users don't have to figure out how to path to this directory themselves.